Content.Shared/Abilities/Goliath/SharedGoliathTentacleSystem.cs

```diff
@@ -0,0 +1,70 @@
+using Content.Shared.Abilities.Goliath.Components;
+using Content.Shared.Directions;
+using Content.Shared.Maps;
+using Content.Shared.Physics;
+using Content.Shared.Popups;
+using Content.Shared.Stunnable;
+using Robust.Shared.Map;
+using Robust.Shared.Map.Components;
+using Robust.Shared.Network;
+using Robust.Shared.Random;
+
+namespace Content.Shared.Abilities.Goliath;
+
+public abstract class SharedGoliathTentacleSystem : EntitySystem
+{
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly INetManager _net = default!;
+    [Dependency] private readonly SharedMapSystem _map = default!;
+    [Dependency] private readonly SharedStunSystem _stun = default!;
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+    [Dependency] private readonly TurfSystem _turf = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<GoliathTentacleUserComponent, GoliathSummonTentacleAction>(OnSummonAction);
+    }
+
+    private void OnSummonAction(Entity<GoliathTentacleUserComponent> ent, ref GoliathSummonTentacleAction args)
+    {
+        if (args.Handled || args.Coords is not { } coords)
+            return;
+
+        // TODO: animation
+
+        _popup.PopupPredicted(Loc.GetString("tentacle-ability-use-popup", ("entity", ent.Owner)), ent, args.Performer, type: PopupType.SmallCaution);
+        _stun.TryStun(ent, TimeSpan.FromSeconds(0.8f), false);
+
+        List<EntityCoordinates> spawnPos = new();
+        spawnPos.Add(coords);
+
+        var dirs = new List<Direction>();
+        dirs.AddRange(ent.Comp.OffsetDirections);
+
+        for (var i = 0; i < 3; i++)
+        {
+            var dir = _random.PickAndTake(dirs);
+            spawnPos.Add(coords.Offset(dir));
+        }
+
+        if (_transform.GetGrid(coords) is not { } grid || !TryComp<MapGridComponent>(grid, out var gridComp))
+            return;
+
+        foreach (var pos in spawnPos)
+        {
+            if (!_map.TryGetTileRef(grid, gridComp, pos, out var tileRef) ||
+                tileRef.IsSpace() ||
+                _turf.IsTileBlocked(tileRef, CollisionGroup.Impassable))
+            {
+                continue;
+            }
+
+            if (_net.IsServer)
+                Spawn(ent.Comp.TentacleId, pos);
+        }
+
+        args.Handled = true;
+    }
```

Comment:
It would be good if this used spells or some other composition way to do "spawn 3 entities in a line" coz it's pretty mid to have to code each of these by hand.